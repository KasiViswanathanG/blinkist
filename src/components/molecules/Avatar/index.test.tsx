import {screen, fireEvent, getByText, render } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'

import AvatarComponent from '.'
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';


describe('Testing the AvatarComponent', () => {
    it('render single name AvatarComponent', () => {
      const avatar = renderer.create(<AvatarComponent userName={'Viswanthan'} color={undefined} icon={undefined}/>).toJSON();
      expect(avatar).toMatchSnapshot();
    });

    it('render double name AvatarComponent', () => {
      const avatar = renderer.create(<AvatarComponent userName={'Kasi Viswanthan'} color={undefined} icon={undefined}/>).toJSON();
      expect(avatar).toMatchSnapshot();
    });

    it('render double name AvatarComponent with deafult color', () => {
      const avatar = renderer.create(<AvatarComponent userName={'Kasi Viswanthan'} color={''} icon={undefined}/>).toJSON();
      expect(avatar).toMatchSnapshot();
    });

    it('render double name AvatarComponent with color', () => {
      const avatar = renderer.create(<AvatarComponent userName={'Kasi Viswanthan'} color={'purple'} icon={undefined}/>).toJSON();
      expect(avatar).toMatchSnapshot();
    });

    it('render double name AvatarComponent with color and icon', () => {
        const avatar = renderer.create(<AvatarComponent userName={'Kasi Viswanthan'} color={'purple'} icon={<KeyboardArrowDownIcon />}/>).toJSON();
        expect(avatar).toMatchSnapshot();
      });
})  