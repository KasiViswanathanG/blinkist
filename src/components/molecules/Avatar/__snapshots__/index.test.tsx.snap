// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the AvatarComponent render double name AvatarComponent 1`] = `
<div
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
  >
    KV
  </div>
  <span
    style={
      Object {
        "marginTop": "9px",
      }
    }
  />
</div>
`;

exports[`Testing the AvatarComponent render double name AvatarComponent with color 1`] = `
<div
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-11v6gn1-MuiAvatar-root"
  >
    KV
  </div>
  <span
    style={
      Object {
        "marginTop": "9px",
      }
    }
  />
</div>
`;

exports[`Testing the AvatarComponent render double name AvatarComponent with color and icon 1`] = `
<div
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-11v6gn1-MuiAvatar-root"
  >
    KV
  </div>
  <span
    style={
      Object {
        "marginTop": "9px",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="KeyboardArrowDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
    </svg>
  </span>
</div>
`;

exports[`Testing the AvatarComponent render double name AvatarComponent with deafult color 1`] = `
<div
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-cieima-MuiAvatar-root"
  >
    KV
  </div>
  <span
    style={
      Object {
        "marginTop": "9px",
      }
    }
  />
</div>
`;

exports[`Testing the AvatarComponent render single name AvatarComponent 1`] = `
<div
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <div
    className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
  >
    V
  </div>
  <span
    style={
      Object {
        "marginTop": "9px",
      }
    }
  />
</div>
`;
